### PR TITLE
Check presence of field, rather than extra smarty variable for whether to display the field

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
https://lab.civicrm.org/dev/core/-/issues/4006

### CODE REVIEW

File: CRM/Contribute/Form/SoftCredit.php [hunk 1]
```diff
@@ -108,16 +108,15 @@ public static function buildQuickForm(&$form) {
...
    * Add PCP fields for the new contribution form and others.
    *
-   * @param CRM_Core_Form &$form
+   * @param CRM_Core_Form $form
    *   The form being built.
    * @param string $suffix
    *   A suffix to add to field names.
    */
-  public static function addPCPFields(&$form, $suffix = '') {
+  public static function addPCPFields($form, $suffix = '') {
```
Comment (mattwire) [inline CRM/Contribute/Form/SoftCredit.php:116]: Are you sure about this? Doesn't it need to be passed by reference?

Comment (eileenmcnaughton) [reply CRM/Contribute/Form/SoftCredit.php:116]: @mattwire yeah pass-by-ref is meaningless for objects (maybe that wasn't true in php 5.1 when they added it)

File: templates/CRM/Contribute/Form/Contribution.tpl [hunk 2]
```diff
@@ -284,7 +284,7 @@
     <!-- end of soft credit -->
 
     <!-- start of PCP -->
-    {if $siteHasPCPs && !$payNow}
+    {if array_key_exists($form, 'pcp_made_through_id') && !$payNow}
```
Comment (demeritcowboy) [inline templates/CRM/Contribute/Form/Contribution.tpl:287]: This looks backwards: `array_key_exists('pcp_made_through_id', $form)`

Comment (demeritcowboy) [reply templates/CRM/Contribute/Form/Contribution.tpl:287]: But also if $suffix isn't blank this wouldn't be true, right?

Comment (demeritcowboy) [reply templates/CRM/Contribute/Form/Contribution.tpl:287]: Hmm but the tpl assumes it always is blank. Civi is weird.

Comment (eileenmcnaughton) [reply templates/CRM/Contribute/Form/Contribution.tpl]: Yep - agree on all counts :-)

**Pull request merged**